SearchParser.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * SearchParser
+ * @param {Array} patterns  Enumerates params and the patterns that match them
+ * @param {boolean} options.returnAllMatches  If true return all matches (not just first)
+ */
+
+'use strict';
+
+module.exports = function SearchParser(patterns, options) {
+    this.patterns = patterns;
+    this.returnAllMatches = options ? options.returnAllMatches : false;
+    /**
+     * Run regex match against supplied search input
+     * @param  {string} input
+     * @return {Object}       Match result, with supplied params as keys
+     *                        and matched strings as values
+     */
+    this.match = function (input) {
```

Comment:
Is there a good reason to define this on the instance? The standard pattern is too declare methods on the prototype.